packages/cms-lib/lib/config/oauth2.js

```diff
@@ -0,0 +1,42 @@
+const { OAUTH_AUTH_METHOD } = require('../constants');
+
+/**
+ * Generates a portalConfig object from previous values and desired updates
+ * @param {object} portalConfig Existing oauth2 portalConfig
+ * @param {object} configUpdates Object containing desired updates
+ */
+const getUpdatedOauthPortalConfig = (portalConfig = {}, configUpdates = {}) => {
+  const auth = {
+    ...portalConfig.auth,
+    ...configUpdates.auth,
+  };
+
+  if (!auth) {
+    throw new Error('No auth data passed to getUpdatedOauthPortalConfig.');
+  }
+
+  if (!auth.tokenInfo) {
+    throw new Error(
+      'No auth.tokenInfo data passed to getUpdatedOauthPortalConfig.'
+    );
+  }
+
+  const config = {
+    ...portalConfig,
+    portalId: configUpdates.portalId,
+    name: configUpdates.name || portalConfig.name,
+    env: configUpdates.env || portalConfig.env,
+    defaultMode: configUpdates.defaultMode || portalConfig.defaultMode,
+    authType: OAUTH_AUTH_METHOD.value,
+    auth,
+  };
+
+  delete config.apiKey;
+  delete config.personalAccessKey;
```

Comment:
Due to how the individual authType update functions are called from `updatePortalConfig`, I don't think we should have destructive operations like this. I think our philosophy when it comes to updating the config is that we should only do destructive operations when a user is going through a prompted flow like `hs auth` or `hs config` (when added).

Reply:
These methods are only accessed from `hs init` and `hs auth`.

The reason for the destruction(`delete`) is to remove any props that get passed in through the existing `portalConfig` and are not necessary for the new `authType` `portalConfig data shape. For example, when using `authType: apikey` and converting to `authType: personalaccesskey` the `apiKey` prop is no longer wanted, so we discard it.

Only the possible leftover keys from other portalConfig data shapes are explicitly deleted.

This allows the following `apikey` data shape...
![image](https://user-images.githubusercontent.com/6472448/74991801-52fdd700-5447-11ea-8816-88d664de3d94.png)
...to be changed properly to `personalaccesskey` by running `hs auth personalaccesskey`...
![image](https://user-images.githubusercontent.com/6472448/74991757-32ce1800-5447-11ea-8bf7-02da940b18f2.png)

My goal here was to create pure functions for `apikey`, `oauth2`, and `personalaccesskey` that take in the existing `portalConfig` as well as the desired `configUpdates` and return a new object with only valid properties for that `authType`, however, allowing as few or as many properties to be updated by passing them as `configUpdates`.

Reply:
> These methods are only accessed from hs init and hs auth.

Because the specialized updaters are called from `updatePortalConfig` based on the `authType`, this is not correct. This means that if for some reason I edit the config file to temporarily switch from using a personalaccesskey to using an api key that `personalAccessKey` will be blown away. I don't now how often that will happen in the real world, but it something that I do quite regularly when testing stuff.

Reply:
Okay I'll remove the `delete` calls to prevent the superfluous props from getting removed